### ISSUE TITLE
no-magic-number don't handle negative zero well

### ISSUE THREAD

**Issue by 7sDream** [palantir/tslint#3840]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.8.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
console.log(-0)
```

with `tslint.json` configuration:

```json
"no-magic-numbers": [true, -2, -1, -0, 0, 1, 2, 3],
```

PS: VSCode shows `Array has duplicate items` problem.

#### Actual behavior

Error:

```
[1, 13]: 'magic numbers' are not allowed
```

#### Expected behavior

No error

**Comment by giladgray** [comment]
fixed in #3903
